tortoise/functions.py

```diff
@@ -59,9 +59,13 @@ def _resolve_field_for_model(
                 function_joins.append(join)
                 field = related_field_meta.basetable[related_field_meta.db_pk_field]
             else:
-                field = table[field_split[0]]
+                field_object = cast(Field, model._meta.fields_map.get(field_split[0]))
+                if field_object.source_field:
+                    field = table[field_object.source_field]
+                else:
+                    field = table[field_split[0]]
                 if self.populate_field_object:
-                    self.field_object = model._meta.fields_map.get(field_split[0], None)
+                    self.field_object = model._meta.fields_map.get(field_split[0])
```

Comment:
If you don't want a default, then isn't using `[]` easier to read? Also, I think, faster. (Mostly because `[]` is builtin, and `.get()` is a method call)

Reply:
Okay :) i will fix it